### PR TITLE
[documentation] LocalPathProvisioner examples are not clear enough

### LINKED ISSUE #360: LocalPathProvisioner: Retain retention policy
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

At this moment `Delete` retention policy is [hardcoded](https://github.com/deckhouse/deckhouse/blob/d1282028c698f24ae2c97084d51ad5e2f880b9b7/modules/031-local-path-provisioner/templates/storage-classes.yaml#L12) into provisioner definition, which makes it impossible to create volumes with `Retain` policy which might be highly required for stateful applications (otherwise why bother at all with module)

### Proposed Solution

From one side ideally, if we will have retention policy property in CRD, but it might require making changes in 3rd party provisioner, so probably to keep things simple it will be better and easier just to have two provisioners

### Additional Information

Did not found how deletion hook is triggered, technically if at moment when hook is triggered we know retention policy we can bypass it to teardown script and make decision whether folder should be delete or not

### LINKED ISSUE #363: proposal: [documentation] LocalPathProvisioner examples are not clear enough
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

Takes me a few days to realize how everything works 🤷‍♂️

Thankfully engineers from the Flant did help with answers and guidance 💪

### Proposed Solution

Example section of LocalPathProvisioner should shout about storage class and have complete usage examples so it will become really obvious how to use it

### Additional Information

#360 - related issue about retention policy

This issue is a template to create and link PR for docs, hopefully it will be accepted 🤞

### PR DESCRIPTION
**Pull request by mac2000**
## Description

Did add some more examples, questions and answers we have struggled with while tried to figure out how everything works

## Why we need it and what problem does it solve?

Not sure if I only one who did not realize all that things from the start, but hopefully it will help future users to get up and running faster

## Changelog entries

```changes
module: local-path-provisioner
type: feature
description: documentation examples added
note: documentation on website should differ and have more content after release
```

---

Fixes #363

### CODE REVIEW

File: modules/031-local-path-provisioner/docs/FAQ.md [hunk 1]
```diff
@@ -27,3 +27,13 @@ prometheus: |
 ```
 
 Wait for the restart of Prometheus Pods.
+
+## Ho to change retention policy?
+
+At moment delete retention policy is hardcoded, and there is no way to change it [issue](https://github.com/deckhouse/deckhouse/issues/360)
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/FAQ.md:33]: Retention policy `Delete` is hardcoded, and there is no way to change it [issue](https://github.com/deckhouse/deckhouse/issues/360).

File: modules/031-local-path-provisioner/docs/FAQ.md [hunk 2]
```diff
@@ -27,3 +27,13 @@ prometheus: |
...
+
+## Why folder not deleted from server after cleanup?
+
+If you do comething like `kubectl delete -f demo.yml` it does delete `LocalPathProvisioner` which is responsible for folder deletion, so in other words there is no one who will be able to run `rm -rf /mnt/kubernetes/demo` for you.
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/FAQ.md:37]: I think this typo is not needed due to absent demo.yml in examples.

File: modules/031-local-path-provisioner/docs/FAQ_RU.md [hunk 3]
```diff
@@ -27,3 +27,13 @@ prometheus: |
 ```
 
 Дождаться переката Pod'ов Prometheus.
+
+## Как изменить политику очистки?
+
+На текущий момент политика удаления зашита в исходные коды и не может быть изменена [issue](https://github.com/deckhouse/deckhouse/issues/360)
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/FAQ_RU.md:33]: Политика удаления установлена в `Delete` и не может быть изменена [issue](https://github.com/deckhouse/deckhouse/issues/360).

File: modules/031-local-path-provisioner/docs/FAQ_RU.md [hunk 4]
```diff
@@ -27,3 +27,13 @@ prometheus: |
...
+
+## Почему папка не была удалена с сервера после очистки?
+
+Если была выполнена команда по типу `kubectl delete -f demo.yml` она удалила все разом, включая `LocalPathProvisioner` который в свою очередь отвечает за фактическое удаление папки, другими словами просто некому выполнить `rm -rf /mnt/kubernetes/demo`
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/FAQ_RU.md:37]: I think this typo is not needed due to absent demo.yml in examples.

File: modules/031-local-path-provisioner/docs/FAQ_RU.md [hunk 5]
```diff
@@ -27,3 +27,13 @@ prometheus: |
...
+
+Для того что бы все сработало как надо, необходимо удалить pod'ы, следом pvc, уже затем модуль `LocalPathProvisioner` увидит эти изменения и очистит папки на сервере.
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/FAQ_RU.md:39]: Папки на диске удаляются `localPathProvisioner` автоматически после удаления pod'ов и pvc.

File: modules/031-local-path-provisioner/docs/FAQ_RU.md [hunk 6]
```diff
@@ -27,3 +27,13 @@ prometheus: |
 ```
 
 Дождаться переката Pod'ов Prometheus.
+
+## Как изменить политику очистки?
+
+На текущий момент политика удаления зашита в исходные коды и не может быть изменена [issue](https://github.com/deckhouse/deckhouse/issues/360)
+
+## Почему папка не была удалена с сервера после очистки?
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/FAQ_RU.md:35]: ## Удаление папок с диска.

File: modules/031-local-path-provisioner/docs/FAQ.md [hunk 7]
```diff
@@ -27,3 +27,13 @@ prometheus: |
 ```
 
 Wait for the restart of Prometheus Pods.
+
+## Ho to change retention policy?
+
+At moment delete retention policy is hardcoded, and there is no way to change it [issue](https://github.com/deckhouse/deckhouse/issues/360)
+
+## Why folder not deleted from server after cleanup?
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/FAQ.md:35]: ## Folder removing from disk.

File: modules/031-local-path-provisioner/docs/README_RU.md [hunk 8]
```diff
@@ -19,4 +19,6 @@ title: "Модуль local-path-provisioner"
   ```
 
 ## Ограничения
-Ограничение на размер диска не поддерживается для локальных томов.
+
+- Ограничение на размер диска не поддерживается для локальных томов.
+- `Delete` политика зашита в исходные коды [issue](https://github.com/deckhouse/deckhouse/issues/360)
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/README_RU.md:24]:
```suggestion
-  Политика по умолчанию - `Delete`, изменить это нельзя [issue](https://github.com/deckhouse/deckhouse/issues/360)
```

File: modules/031-local-path-provisioner/docs/USAGE.md [hunk 9]
```diff
@@ -14,3 +14,10 @@ spec:
   - system
   path: "/opt/local-path-provisioner"
 ```
+
+Notes:
+
+- This example will create `localpath-system` storage class which **must** be used by pods for everything to work
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/USAGE.md:20]:
```suggestion
- This example will create `localpath-system` storage class which be used by prometheus pods to store data.
```

File: modules/031-local-path-provisioner/docs/USAGE.md [hunk 10]
```diff
@@ -14,3 +14,10 @@ spec:
...
+- Volumes created by provisioner will have delete retention policy which is hardcoded ([issue](https://github.com/deckhouse/deckhouse/issues/360))
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/USAGE.md:21]:
```suggestion
- Volumes created by provisioner will have default delete retention policy which is hardcoded ([issue](https://github.com/deckhouse/deckhouse/issues/360)).
```

File: modules/031-local-path-provisioner/docs/USAGE.md [hunk 11]
```diff
@@ -14,3 +14,10 @@ spec:
...
+- If provisioner will be delete before claims folders will not be deleted from node
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/USAGE.md:22]:
```suggestion
- If the provisioner will be deleted before claims folders will not be deleted from node.
```

File: modules/031-local-path-provisioner/docs/USAGE.md [hunk 12]
```diff
@@ -14,3 +14,10 @@ spec:
...
+- Note that in example `system` node is used which probably will have some taints so pods **must** have corresponding tolerations
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/USAGE.md:23]: this isn't needed due to prometheus has tolerations for system nodes by default.

File: modules/031-local-path-provisioner/docs/USAGE_RU.md [hunk 13]
```diff
@@ -14,3 +14,10 @@ spec:
   - system
   path: "/opt/local-path-provisioner"
 ```
+
+Примечания:
+
+- Этот пример создаст `localpath-system` класс (`storage class`) который **должен** быть использован в pod'ах что бы все заработало
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/USAGE_RU.md:20]:
```suggestion
- Этот пример создаст `localpath-system` storage-class который  будет использоваться для хранения данных в pod'ах prometheus.
```

File: modules/031-local-path-provisioner/docs/USAGE_RU.md [hunk 14]
```diff
@@ -14,3 +14,10 @@ spec:
...
+- Все создаваемые хранилища будут иметь политику очистки `Delete` ([issue](https://github.com/deckhouse/deckhouse/issues/360))
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/USAGE_RU.md:21]:
```suggestion
- Все создаваемые хранилища будут иметь политику очистки по-умолчанию `Delete`, которую нельзя изменить ([issue](https://github.com/deckhouse/deckhouse/issues/360)).
```

File: modules/031-local-path-provisioner/docs/USAGE_RU.md [hunk 15]
```diff
@@ -14,3 +14,10 @@ spec:
...
+- Если этот объект будет удален раньше чем объекты его использующие, папки с сервера удалены не будут
+- Обратите внимание - в примере предпологается создание папок на системных нодах, которые скорее всего имеют ряд ограничителей (taints), а как следствие pod'ы **должны** иметь соотв. tolerations
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/USAGE_RU.md:23]: this isn't needed due to prometheus has tolerations for system nodes by default.

File: modules/031-local-path-provisioner/docs/USAGE_RU.md [hunk 16]
```diff
@@ -14,3 +14,10 @@ spec:
   - system
   path: "/opt/local-path-provisioner"
 ```
+
+Примечания:
+
+- Этот пример создаст `localpath-system` класс (`storage class`) который **должен** быть использован в pod'ах что бы все заработало
+- Все создаваемые хранилища будут иметь политику очистки `Delete` ([issue](https://github.com/deckhouse/deckhouse/issues/360))
+- Если этот объект будет удален раньше чем объекты его использующие, папки с сервера удалены не будут
```
Comment (RomanenkoDenys) [inline modules/031-local-path-provisioner/docs/USAGE_RU.md:22]:
```suggestion
- Если `provisioner` будет удален раньше чем объекты его использующие, папки с сервера удалены не будут.
```

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]

**Comment by nabokihms** [comment]
Denys and Mac2000 reached the agreement, that the problem with retention is fixed by this PR https://github.com/deckhouse/deckhouse/pull/561

As no more work is planned for this PR, I will close it. Feel free to reopen if needed.

**Pull request closed without merging**